examples/zone-printer/src/main.go

```diff
@@ -38,13 +56,74 @@ func main() {
 		log.Printf("info: determined zone: %q", zone)
 	}
 
-	log.Println("starting to listen on port 80")
-	http.HandleFunc("/", handle)
-	err := http.ListenAndServe(":80", nil)
-	log.Fatal(err)
+	finished := make(chan bool)
+
+	log.Printf("starting to listen on port %d", mainPort)
+	mainServer := http.NewServeMux()
+	mainServer.HandleFunc("/", handlePing)
+	mainServer.HandleFunc("/ping", handlePing)
+	mainServer.HandleFunc("/location", handleLocation)
+	mainServer.HandleFunc("/toggleKill", handleKill)
+	go func() {
+		portStr := fmt.Sprintf(":%d", mainPort)
+		log.Fatal(http.ListenAndServe(portStr, mainServer))
+	}()
+
+	log.Printf("starting to listen on port %d", *healthCheckPort)
+	hcServer := http.NewServeMux()
+	hcServer.HandleFunc(*healthCheckPath, handleHealthCheck)
+	go func() {
+		portStr := fmt.Sprintf(":%d", *healthCheckPort)
+		log.Fatal(http.ListenAndServe(portStr, hcServer))
+	}()
+
+	<-finished
+}
+
+type pingResponse struct {
+	Hostname string
+	Version string
+	GCPZone string
+	Backend string
+}
+
+func handlePing(w http.ResponseWriter, r *http.Request) {
+	podHostname, err := os.Hostname()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	pr := &pingResponse{Hostname: r.Host, Version: *version, GCPZone: computeZone, Backend: podHostname}
+	j, err := json.Marshal(pr)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(j)
+}
+
+func handleKill(w http.ResponseWriter, r *http.Request) {
+	lock.Lock()
+	defer lock.Unlock()
+
+	killed = !killed
+	fmt.Fprintf(w, "Successfully toggled kill status")
+}
+
+func handleHealthCheck(w http.ResponseWriter, r *http.Request) {
+	lock.Lock()
+	defer lock.Unlock()
```

Comment:
bool’s are atomic so you actually don't need locks around them.

if you're really interested in locking you can prob. use RWLock and gain a RLock here.

Reply:
Really? I don't see any documentation on that. Used RWMutex.